promql/functions.go

```diff
@@ -1386,6 +1386,9 @@ func (ev *evaluator) evalLabelJoin(args parser.Expressions) (parser.Value, annot
 		}
 		srcLabels[i-3] = src
 	}
+	if !model.LabelName(dst).IsValid() {
```

Comment:
I would validate the destination before the source arguments, to save work in the case of error.